modules/querier/querier.go

```diff
@@ -221,7 +221,7 @@ func (q *Querier) stopping(_ error) error {
 }
 
 // FindTraceByID implements tempopb.Querier.
-func (q *Querier) FindTraceByID(ctx context.Context, req *tempopb.TraceByIDRequest, timeStart int64, timeEnd int64) (*tempopb.TraceByIDResponse, error) {
+func (q *Querier) FindTraceByID(ctx context.Context, req *tempopb.TraceByIDRequest, timeStart int64, timeEnd int64, allowPartialTrace bool) (*tempopb.TraceByIDResponse, error) {
```

Comment:
the querier checks both backend blocks and queries the ingesters. as you've found there are multiple layers where we fail the query if the trace exceeds max size. we need to find a way to bubble up from those places that this is occurred so the frontend knows to communicate back to the caller that this is a partial trace.

technically it should be impossible for a trace to exceed the max size in one block. this should only occur when recombining and higher levels. perhaps we drop that code that tests this?

Reply:
> the querier checks both backend blocks and queries the ingesters. as you've found there are multiple layers where we fail the query if the trace exceeds max size. we need to find a way to bubble up from those places that this is occurred so the frontend knows to communicate back to the caller that this is a partial trace.

We could use the TraceById Response for that. So we can wrap the trace there.

> technically it should be impossible for a trace to exceed the max size in one block. this should only occur when recombining and higher levels. perhaps we drop that code that tests this?

I would love to know why is that 🙇. I always favor deleting code :)

Reply:
I know kung fu now :P